ios/Exponent/Versioned/Core/UniversalModules/EXScopedModuleRegistryAdapter.m

```diff
@@ -170,8 +170,11 @@ - (UMModuleRegistry *)moduleRegistryForParams:(NSDictionary *)params forExperien
 #endif
   
 #if __has_include(<EXNotifications/EXNotificationCategoriesModule.h>)
-  EXScopedNotificationCategoriesModule *categoriesModule = [[EXScopedNotificationCategoriesModule alloc] initWithExperienceId:experienceId];
-  [moduleRegistry registerExportedModule:categoriesModule];
+  // only override in Expo Go
```

Comment:
Took a stab at that [here](https://github.com/expo/expo/pull/11710), but I think I prefer that we just expose a JS method that does this for you in SDK 41 and beyond. This would be simpler for a few reasons:

- that way, `EXScopedNotificationCategoriesModule` would _only_ used in Expo Go, and standalone & bare apps would rely on the same code
- People ejecting to the bare workflow can use the same JS method to migrate their own categories post-ejection
- the migration code won't be being run multiple times (to be clear- there's no _problem_ with running the migration method from #11710 again and again, it just does nothing) 

The drawback, however, would be that we'd expose our scoping implementation and letting that complication leak into Expo developers' lives (+ they'd have to read the changelog and know to call `Notifications.migrateCategories()` or whatever)

Reply:
Those tradeoffs make sense to me. With the API, developers don't need to worry about how the categories are scoped, but they would need to know that there is scoping in older SDKs. Thinking through your points, which I thought were good, I lean towards doing the descoping natively:

> EXScopedNotificationCategoriesModule would only used in Expo Go, and standalone & bare apps would rely on the same code

We could address this by putting the descoping logic in a class that is excluded from Expo Go but used everywhere else. This would give us parity between standalone & bare apps and keep the logic pretty minimal.

> People ejecting to the bare workflow can use the same JS method to migrate their own categories post-ejection (instead of needing to upgrade to SDK 41 first, then ejecting)

This makes sense. As I understand it, the main benefit of letting SDK 40 apps do this migration is that they'd be able to use unscoped categories (if we added server support). My sense is that it's fine to require an SDK upgrade since we aren't taking away any functionality from old SDKs -- upgrading gives you new functionality.

> the migration code won't be being run multiple times

Developers would need to keep some state if they wanted to call the descoping method from JS just once. My sense is we should profile how long it takes to look up all of the notification categories and iterate through each one without renaming them. That is, how expensive would it be to run the descoping logic on every app launch? It might be trivial, say 0.05 ms, and be fine to run each time. Or maybe it takes 3~4 ms and be long enough that we might want to keep a small piece of state around (e.g. in NSUserDefaults) to remember that we've already descoped the categories.

Reply:
Great call to profile the descoping logic! Ran these tests looping through 20 notification categories (I felt like this is a safe overestimate):
- To run the descoping logic **and** actually descope took 1.2ms
- To run the descoping logic after having **already** descoped previously took 0.1ms ⏩ 

This makes sense, since most of the time in the first case was spent in serializing the existing category and then creating a new one. Result: seems safe to run the descoping logic on every app launch

Reply:
That's great to hear. This sounds like a good approach -- adding 0.1ms (which can run while JS is being read from disk and parsed and run, potentially) is negligible and worth it for having notification categories be consistently correct.

We will need to make sure that this code runs only at launch, and only outside of Expo Go.

Reply:
Since we _do_ still need to run a migration for Expo Go:

> Expo Go would need to change delimiters from - to ??

I think I'm going to backtrack and say using `EXScopedNotificationCategoriesModule` in both Expo Go and standalone contexts for now is fine. In [both cases we need to run a migration](https://github.com/expo/expo/pull/11710/files#diff-13fa1ae9cacf843f9cfd44c3967b6bf860ecc5e0356ffefde4647e75b552dbb0R20-R28):

| Scoping? | in Expo Go  | in Standalone |
| ------------- | ------------- | ------------- |
| Categories are scoped with `-` delimiter  | migrate to `/` delimiter  | migrate to unscoped |
| Categories are scoped with `/` delimiter  | migration already done ✅  | Impossible scenario A  |
| Categories are unscoped  | Impossible scenario B  | migration already done ✅  |

> **Impossible scenario A**: categories have thus far been scoped with `-` delimiter in standalones, and we never migrate to the `/` delimiter in standalones, so this would never be the case.

> **Impossible scenario B**: categories will never be unscoped in Expo Go, since we've scoped them since the feature's release. Even the home app's categories itself (not that we create or use any) are technically scoped under that experience.

Reply:
Even though we need to run a migration in both cases, the migrations are different and don't need to be part of the same class, let alone a Module class.

Reply:
extracted out to it's own class